theano/tensor/opt.py

```diff
@@ -5271,6 +5271,16 @@ def local_intdiv_by_one(node):
             return [node.inputs[0].astype(node.outputs[0].dtype)]
 
 
+@register_canonicalize
+@gof.local_optimizer([T.int_div, T.true_div, T.floor_div])
+def local_zero_div(node):
+    """0 / x -> 0
+    """
+    if node.op in [T.int_div, T.true_div, T.floor_div]:
+        if local_mul_canonizer.get_constant(node.inputs[0]) == 0:
```

Comment:
`local_mul_canonizer.get_constant` is a wrapper that does the correct thing already (I copied it from one of the other optimizers, not sure why they don't call `get_scalar_constant_value` directly),